### DISCUSSION TITLE
ioda2nc, netcdf dimension name

### DISCUSSION THREAD

**Discussion by fearon10** [dtcenter/METplus#2290] [File I/O] [answered]
Hi There,

I'm working with ioda formatted files (version 3 I believe, ncdump of this file is below) where the dimension name "Location" is used rather than "nlocs". However, ioda2nc is looking "nlocs". Perhap ioda2nc does not work ioda version 3 yet and, if that's the case, is there a possible workaround? I tried using nco to rename the "Location" dimension to "nlocs" but the nco utility does not successfully read ioda netcdf. 

cmd: /usr/local/bin/ioda2nc scatterometer_ascat.nc4 test.nc -config IODA2NCConfig_default
DEBUG 1: Default Config File: /usr/local/share/met/config/IODA2NCConfig_default
DEBUG 1: User Config File: IODA2NCConfig_default
DEBUG 1: Creating NetCDF File:          test.nc
DEBUG 1: Processing IODA File:  scatterometer_ascat.nc4
ERROR  :
ERROR  : get_dim() -> can't find NetCDF dimension "nlocs".
ERROR  :

Thank you for the assistance.
Matt

-----------------
ncdump -h scatterometer_ascat.nc4

netcdf scatterometer_ascat {
dimensions:
        Location = 6609 ;
variables:
        int Location(Location) ;
                Location:suggested_chunk_dim = 6609LL ;

// global attributes:
                string :_ioda_layout = "ObsGroup" ;
                :_ioda_layout_version = 0 ;
                string :converter = "def_scatterometer_base.py" ;
                string :sensor = "scatterometer_ascat" ;
                string :platform = "scatterometer_ascat" ;

group: MetaData {
  variables:
        int64 dateTime(Location) ;
                dateTime:_FillValue = -9223372036854775806LL ;
                string dateTime:units = "seconds since 1970-01-01T00:00:00Z" ;
                string dateTime:coordinates = "longitude latitude" ;
        int insty(Location) ;
                insty:_FillValue = -2147483647 ;
                string insty:coordinates = "longitude latitude" ;
        float latitude(Location) ;
                latitude:_FillValue = -3.368795e+38f ;
                string latitude:coordinates = "longitude latitude" ;
                string latitude:units = "degrees_north" ;
        float longitude(Location) ;
                longitude:_FillValue = -3.368795e+38f ;
                string longitude:coordinates = "longitude latitude" ;
                string longitude:units = "degrees_east" ;
        float stationElevation(Location) ;
                stationElevation:_FillValue = -3.368795e+38f ;
                string stationElevation:units = "m" ;
                string stationElevation:coordinates = "longitude latitude" ;
        string stationIdentification(Location) ;
                string stationIdentification:_FillValue = "" ;
                string stationIdentification:coordinates = "longitude latitude" ;
  } // group MetaData

group: ObsError {
  variables:
        float windEastward(Location) ;
                windEastward:_FillValue = -3.368795e+38f ;
                string windEastward:coordinates = "longitude latitude" ;
                string windEastward:units = "m/sec" ;
        float windNorthward(Location) ;
                windNorthward:_FillValue = -3.368795e+38f ;
                string windNorthward:coordinates = "longitude latitude" ;
                string windNorthward:units = "m/sec" ;
  } // group ObsError

group: ObsValue {
  variables:
        float windEastward(Location) ;
                windEastward:_FillValue = -3.368795e+38f ;
                string windEastward:coordinates = "longitude latitude" ;
                string windEastward:units = "m/sec" ;
        float windNorthward(Location) ;
                windNorthward:_FillValue = -3.368795e+38f ;
                string windNorthward:coordinates = "longitude latitude" ;
                string windNorthward:units = "m/sec" ;
  } // group ObsValue

group: PreQC {
  variables:
        int windEastward(Location) ;
                windEastward:_FillValue = -2147483647 ;
                string windEastward:coordinates = "longitude latitude" ;
                string windEastward:units = "m/sec" ;
        int windNorthward(Location) ;
                windNorthward:_FillValue = -2147483647 ;
                string windNorthward:coordinates = "longitude latitude" ;
                string windNorthward:units = "m/sec" ;
  } // group PreQC
}

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. I did a little digging into the C library on the ioda2nc capability and found your assumption is correct; [we currently look for the nlocs variable](https://github.com/dtcenter/MET/blob/3f0411b65cef5719bf11b6fb75f7e8413fe633f0/src/tools/other/ioda2nc/ioda2nc.cc#L438) and if it's unavailable, MET errors out. With the new version of IODA output utilizing Location over nlocs, we'll need to update our support.
I've gone ahead and created [this MET issue detailing your discovery](https://github.com/dtcenter/MET/issues/2640) and calling for full version 3 IODA output to be supported in the ioda2nc tool.
If there are any other important changes you saw between version 2 and 3 that you want to inform the MET team about, please use that Github issue to let us know so we can provide you with the appropriate support.

**Reply by fearon10** [reply]
Thank you @j-opatz. I have not found any others changes/issues between version 2 and 3. If I do, I'll share those.
